instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/logging/OpenTelemetryAppenderAutoConfiguration.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.spring.autoconfigure.logging;
+
+import io.opentelemetry.api.OpenTelemetry;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.event.ApplicationPreparedEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@SuppressWarnings("OtelPrivateConstructorForUtilityClass")
+@ConditionalOnBean(OpenTelemetry.class)
+public class OpenTelemetryAppenderAutoConfiguration {
+
+  @Bean
+  @ConditionalOnProperty(
+      prefix = "otel.springboot.log4j-appender",
+      name = "enabled",
+      matchIfMissing = true)
+  @ConditionalOnClass({
+    io.opentelemetry.instrumentation.log4j.appender.v2_17.OpenTelemetryAppender.class
+  })
+  ApplicationListener<ApplicationPreparedEvent> log4jOtelAppenderInitializer(
+      OpenTelemetry openTelemetry) {
+    return event -> {
+      io.opentelemetry.instrumentation.log4j.appender.v2_17.OpenTelemetryAppender.install(
+          openTelemetry);
+    };
+  }
+
+  @Bean
+  @ConditionalOnProperty(
```

Comment:
It is enabled by default; in case the property is absent it still gets evaluated to `true`. I added it to make it possible to disable the appender auto configuration.

Reply:
Sorry, I was not rather explicit. My point is that I don't see the use case for which the user would like to disable the automatic injection of the OpenTelemetry instance into the OTel appender. I would prefer to remove the property if we don't have a use case in mind.